src/nn_dot_product matrix.c

```diff
@@ -0,0 +1,25 @@
+#define NN_DOT_PRODUCT_MATRIX_C
```

Comment:
Why this? Also the file name has a space character.

Reply:
Just a habit. We had some preprocessing magic that needed these defines. I'm sure Bjarne Stroustrop has something to say. I'll remove it though. And I'll fix the filename.